metpy/xarray.py

```diff
@@ -194,28 +194,28 @@ def _resolve_axis_conflict(self, axis, coord_lists):
                       + 'for variable "' + self._data_array.name + '".')
         coord_lists[axis] = []
 
-    def _parse_coordinates(self):
-        """Parse the coordinates for this variable to identify coordinate types."""
-        # Only parse if _metpy_axis is not present
-        if not any('_metpy_axis' in coord_var.attrs
-                   for coord_var in self._data_array.coords.values()):
-            self.assign_coordinates(self._generate_coordinate_map())
-
     def _metpy_axis_search(self, cf_axis):
-        """Search for a cf_axis in the _metpy_axis attribute on the coordinates."""
-        for coord_var in self._data_array.coords.values():
+        """Search for cached _metpy_axis attribute on the coordinates, otherwise parse."""
+        # Search for coord with proper _metpy_axis
+        coords = self._data_array.coords.values()
```

Comment:
Was there a reason to break apart this line? Otherwise, seems like extra noise in the diff.

Reply:
Yes, with the change below, `self._data_array.coords.values()` gets used a second time, so I figured assigning it to a variable would be more concise. Though, if you'd want me to use the full `self._data_array.coords.values()` expression in both places, I can do that too.

Reply:
I swear I looked for that and didn't find it being used. Now that I put it out there publicly, I had no problem finding it. 🐑